model-server/src/main/kotlin/org/modelix/model/server/handlers/ModelReplicationServer.kt

```diff
@@ -290,6 +301,15 @@ class ModelReplicationServer(val repositoriesManager: RepositoriesManager) {
             }
         }
 
+        get<Paths.postRepositoryBranchQueryWarmup> { parameters ->
+            val branchRef = RepositoryId(parameters.repository).getBranchReference(parameters.branch)
+            val version = repositoriesManager.getVersion(branchRef)
+            withContext(Dispatchers.IO) {
+                inMemoryModel.getModel(version!!.getTree())
+            }
+            call.respondText("Version $version loaded into memory")
+        }
```

Comment:
Depending on how long this might take, especially with a potentially IO-limited database from the cloud, we might get into issues with exceeding the default 30 seconds timeout again.

Reply:
The `inMemoryModel.getModel` call continues even if the client disconnects, because there are no coroutine usages inside of it that would allow the dispatcher to interrupt it. This warmup endpoint can be used in a readiness probe and eventually it will succeed without a delay.

Reply:
Even if that's not a problem for the effect of this operation, having to deal with timeouts is tedious. What about splitting this into two operations. One that starts loading and one that can be used to query if loading finished?

Reply:
The endpoint is intended to be used by a readiness probe. I don't understand how splitting it into two endpoint would make that easier to use.

Reply:
We already have a health endpoint. In k8s, we can only define a single http endpoint for probes and I would also want the health endpoint be usable in addition to fetching the model into memory. So maybe then this should use a similar pattern for specifying checks to execute via query parameters?